src/solvers/flattening/bv_pointers.cpp

```diff
@@ -172,8 +172,38 @@ literalt bv_pointerst::convert_rest(const exprt &expr)
       const bvt &bv0=convert_bv(operands[0]);
       const bvt &bv1=convert_bv(operands[1]);
 
-      return bv_utils.rel(
-        bv0, expr.id(), bv1, bv_utilst::representationt::UNSIGNED);
+      const pointer_typet &type0 = to_pointer_type(operands[0].type());
+      bvt offset_bv0 = offset_literals(bv0, type0);
+
+      const pointer_typet &type1 = to_pointer_type(operands[1].type());
+      bvt offset_bv1 = offset_literals(bv1, type1);
+
+      // comparison over pointers to distinct objects is undefined and yields a
+      // non-deterministic result; alternatively, we could do a comparison over
+      // the integer representation of a pointer
+
+      // do the same-object-test via an expression as this may permit re-using
+      // already cached encodings of the equality test
+      const exprt same_object = ::same_object(operands[0], operands[1]);
+      const bvt &same_object_bv = convert_bv(same_object);
+      CHECK_RETURN(same_object_bv.size() == 1);
+
```

Comment:
Simply use `convert` instead of `convert_bv`?

Reply:
Done. I'll push a new version once I've also fixed the SMT back-end.